test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportDataWithSenderAndReceiverSource.cs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Modules.Test.TestResultCoordinator.Reports.DirectMethod
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    class DirectMethodReportDataWithSenderAndReceiverSource
+    {
+        public static IEnumerable<object[]> GetCreateReportData =>
+            // See TestCreateReportAsync for parameters names
+            new List<object[]>
+            {
+                new object[]
+                {
+                    // NetworkOnSuccess test
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    new int[] { 200, 200, 200, 200, 200, 200, 200 },
+                    new DateTime[]
+                    {
+                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 21, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 7, 0, 0, 0, 0, 0, 0, 0, true
+                },
+                new object[]
+                {
+                    // NetworkOffSuccess test
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    new[] { "1", "2", "4", "5", "6", "7" },
+                    new int[] { 200, 200, 500, 200, 200, 200, 200 },
+                    new DateTime[]
+                    {
+                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 16, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 6, 1, 0, 0, 0, 0, 0, 0, true
+                },
+                new object[]
+                {
+                    // NetworkOnToleratedSuccess test
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    new[] { "1", "2", "4", "5", "6", "7" },
+                    new int[] { 200, 200, 500, 200, 200, 200, 200 },
```

Comment:
can you use HttpStatusCode.Ok and HttpStatusCode.InternalServerError; and use new List<HttpStatusCode>.